src/components/searchBar/SearchBar.tsx

```diff
@@ -0,0 +1,44 @@
+import styles from './searchBar.module.scss';
+import { ChangeEvent, Dispatch, FC, useEffect, useState } from 'react';
+import { useSearch } from '../../hooks/useSearch.ts';
+import { useDebounce } from '../../hooks/useDebounce.tsx';
+import { TUser } from '../../types/user.ts';
+import { TRepositorie } from '../../types/repositories.ts';
+import SearchImge from '../../assets/search.svg';
+
+interface PropsSearchBar {
+  setUserData: Dispatch<React.SetStateAction<TUser | undefined>>;
+  setRepositories: Dispatch<React.SetStateAction<TRepositorie[] | undefined>>;
+}
```

Comment:
```suggestion
type Props = {
  setUserData: Dispatch<React.SetStateAction<TUser | undefined>>;
  setRepositories: Dispatch<React.SetStateAction<TRepositorie[] | undefined>>;
}
```
use `type Props` for components
try instead of this Dispatch<React.SetStateAction<TUser | undefined>> this (value: TUser | undefined) => void

Reply:
Corrected